Dockerfile

```diff
@@ -1,101 +1,66 @@
 #### Common Build Image Stage ####
 
-FROM ruby:2.7.1-alpine AS common-build-env
+FROM ruby:2.7.1-alpine AS common-build
 
-ARG APP_HOME=/app
-ARG BUILD_PACKAGES="build-base"
-ARG DEV_PACKAGES="postgresql-dev git nodejs yarn graphviz ttf-ubuntu-font-family"
-ARG RUBY_PACKAGES="tzdata"
+ARG BUILD_DEPS="git gcc libc-dev make nodejs yarn postgresql-dev graphviz-dev graphviz-doc build-base libxml2-dev libxslt-dev ttf-ubuntu-font-family"
 ARG bundleWithout=""
 
-ENV BUNDLER_VERSION="2.1.4" \
-    BUNDLE_WITHOUT=${bundleWithout} \
-    WKHTMLTOPDF_GEM=wkhtmltopdf-binary-edge-alpine
-
-RUN apk update && \
-    apk upgrade && \
-    apk add --update --no-cache $BUILD_PACKAGES $DEV_PACKAGES $RUBY_PACKAGES && \
-    gem update --system && \
-    find / -wholename '*default/bundler-*.gemspec' -delete && \
-    rm /usr/local/bin/bundle && \
-    rm /usr/local/bin/bundler && \
-    gem install bundler -v $BUNDLER_VERSION
-
-WORKDIR $APP_HOME
-
-COPY Gemfile Gemfile.lock package.json yarn.lock ./
-
-RUN bundle install && \
-    yarn install
-
-
-#### Dev build builds on common build env image ####
-
-FROM common-build-env AS dev-build
-ARG APP_HOME=/app
-ENV RAILS_ENV=development
-
-WORKDIR $APP_HOME
-
-COPY . .
-
-CMD bundle exec rails db:migrate && bundle exec rails server -b 0.0.0.0
-
-
-#### Prod minify stage builds the app on the common build env image ####
-
-FROM common-build-env AS prod-minify
-
-ARG APP_HOME=/app
-
 # These variables are required for running Rails processes like assets:precompile
-ENV RAILS_ENV=production \
+ENV BUNDLE_WITHOUT=${bundleWithout} \
```

Comment:
Do we use `BUNDLE_WITHOUT` anywhere?

Reply:
bundler uses `BUNDLE_WITHOUT` as an alternative to `--without` switch
https://bundler.io/v1.16/bundle_config.html#CONFIGURATION-KEYS

Reply:
Nice! TIL